src/rust/engine/src/context.rs

```diff
@@ -59,6 +59,7 @@ impl Core {
     ignore_patterns: &[String],
     work_dir: PathBuf,
     local_store_dir: PathBuf,
+    enable_remoting: bool,
```

Comment:
@hrfuller : Do we want to pass this down to the rust code, or just shortcircuit at a higher level on the python side?

Reply:
Wouldn't that require a major rewrite of the Python<->Rust remote interop? It seems like the Rust code is already the main place that deals with how to handle remoting, e.g. speculation, and I would be concerned to start leaking that abstraction into the Python side.

As someone who wasn't worked with the Rust side as much, I thought this current boundary made a lot of sense and I was impressed this change took so few lines.

Reply:
> Wouldn't that require a major rewrite of the Python<->Rust remote interop?

No... it would mean having the python side pass an empty list of servers if `!remote_execution`.

Reply:
I think it's better to be explicit about flags, and the 1:1 relationship between them at the rust level, even at the cost of more arguments passed through the interface.